listenbrainz/webserver/static/js/src/listens/ListenCard.tsx

```diff
@@ -46,6 +46,42 @@ export default class ListenCard extends React.Component<
   ListenCardProps,
   ListenCardState
 > {
+  static preciseTimestamp = (listen: Listen): string => {
```

Comment:
We use formats other than Listen in other parts fo the website, for which we'll also want to use this method (once it's in utils.tsx).

In order to make it work everywhere, we'll want to pass just a timestamp argument rather than the whole listen object.

This `listen.listened_at_iso || listen.listened_at` business should be outside of this method consequently.